components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+                paramMap.put(UAEPassAuthenticatorConstants.UAE.OAUTH2_PARAM_STATE, sessionID);
+
+                String envUAEPass = getUAEPassEnvironment(context);
+                String logoutUrl = getLogoutUrl(envUAEPass);
+                logoutUrl = FrameworkUtils.buildURLWithQueryParams(logoutUrl, paramMap);
+                logoutUrl = modifyLogoutUrl(logoutUrl);
+                response.sendRedirect(logoutUrl);
+
+            } catch (IOException e) {
+                LOG.error("Error in initiate logout URI build : ", e);
+                String idpName = context.getExternalIdP().getName();
+                String tenantDomain = context.getTenantDomain();
+                throw new LogoutFailedException("Error occurred while initiating the logout request to IdP: " + idpName
+                        + " of tenantDomain: " + tenantDomain, e);
+            }
+        } else{
+            super.initiateLogoutRequest(request, response, context);
+        }
+    }
+
+
+    /**
+     *
+     * @param request
+     * @param response
+     * @param context
+     */
+    @Override
+    protected void processLogoutResponse(HttpServletRequest request, HttpServletResponse response,
+                                         AuthenticationContext context) {
+
+        if (LOG.isDebugEnabled()) {
+            if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                LOG.debug("Handled logout response from service provider " + request.getParameter("sp") +
+                        " in tenant domain " + IdentityTenantUtil.getTenantDomainFromContext());
+            } else {
+                LOG.debug("Handled logout response from service provider " + request.getParameter("sp") +
+                        " in tenant domain " + request.getParameter("tenantDomain"));
+            }
+        }
+    }
+
+
+    /**
+     * This method is used add additional parameters along with the authorize request.
+     *
+     * @param authenticatorProperties
+     * @param loginPage
+     * @return authzUrl
+     */
+    public String processAdditionalQueryParamSeperation(Map<String, String> authenticatorProperties, String loginPage)
+        throws  UnsupportedEncodingException{
+
+        String additionalQueryParams = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.QUERY_PARAMS);
+        String[] retval;
+
+        if(additionalQueryParams.contains(",")){
+            retval = additionalQueryParams.split(",");
+        } else {
+            retval = additionalQueryParams.split("&");
+        }
+
+        String[] keyValuePairs;
+        Map<String, String> paramMap = new HashMap<>();
+
+        for (int i = 0; i < retval.length; i++) {
+            keyValuePairs = (retval[i]).split("=");
+            paramMap.put(keyValuePairs[0], keyValuePairs[1]);
+        }
+        String finalAuthzUrl = null;
+        try {
+            String authzUrl = FrameworkUtils.buildURLWithQueryParams(loginPage, paramMap);
```

Comment:
```suggestion
            String authzUrl = FrameworkUtils.buildURLWithQueryParams(loginPage, paramMap).toLowerCase();
```

If no other issues, shall we call `toLowerCase()` when initialing the `authzUrl` so that we don't have to call `toLowerCase` twise

Reply:
Removed toLowerCase() function since not necessary.